Sofia Project Berry Box

solen80a opened the pull request:
Please review my project Berry Box.
https://berrybox.netlify.app/

sofialennbom committed:
starting a new project

sofialennbom committed:
Added Flexbox and CSS Grid

sofialennbom committed:
added font, colors and other styling

sofialennbom committed:
Added comment to html and css. Cleaned some code.

sofialennbom committed:
added a hamburger menu, changed header from flex to grid, footer: display none only on home, about and contact.

sofialennbom committed:
added form to order and book a pickup date

sofialennbom committed:
Added hero image in header, boxed in header text.

sofialennbom committed:
Made .header-text a flexbox.

sofialennbom committed:
added styling to form.

sofialennbom committed:
Fixed some styling for form

sofialennbom committed:
added new webp img and seperated articles from form and aligned them.

sofialennbom committed:
Added an hero video and fixed the formatting on form.

sofialennbom committed:
Fixed font size for all screens and created a .js file. Created html for hamburger.

sofialennbom committed:
Hamburger menu opens and closes.

sofialennbom committed:
Fixed some styling on hamburger.

sofialennbom committed:
Fixed formatting for hamburger mobile/tablet.

sofialennbom committed:
Adde links to all navigation and hamburger menu.

sofialennbom committed:
Adjused navigation header for desktop and widescreen, hidden hambuger and make sure the navigation is visible all time. Also changed the p color for navigation header and footer.

sofialennbom committed:
Adjusted form and contact card. Removed hero video.

tildetilde reviewed (commented):
Code Review

**Strengths:**
– Great HTML structure—it's easy to follow, and semantics are well used. I especially appreciate the use of <main>.
– Nice touch including dates in your form!
– Linking the social media icons to actual profiles is a great detail.
– The hamburger menu looks great! It matches the social media icons well, and it's a nice touch that it links to the contact section.
– Using CSS Grid for the articles is a smart approach!
– The responsive design works well for screens larger than 768px.

**Things to Consider:**
1. Even though the header visually resembles a hero section, it might be worth explicitly defining it as such in HTML (e.g., using a dedicated <section class="hero"> with a <div class="hero-image">). This would also allow for adding an alt attribute, which isn't possible in CSS.

2. In Chrome (but not Safari), the hamburger menu takes over the entire screen at screen sizes below 769px, even without clicking on it. It seems the issue is caused by .active being present on page load. The media query @media (min-width: 501px) and (max-width: 768px) includes right: -768px, which positions the menu off-screen. However, since .active is already present, left: 0; is applied immediately, making the menu visible by default. Removing .active from the initial HTML perhaps could fix this.

Current:


```html
<div id="hamburger-menu-closed" class="hamburger-menu-closed active">
    <img id="hamburger-image" class="hamburger-image" src="./images/hamburger_filled.png" alt="hamburger-image">
</div>
```

Suggested Fix:

```html
<div id="hamburger-menu-closed" class="hamburger-menu-closed">
    <img id="hamburger-image" class="hamburger-image" src="./images/hamburger_filled.png" alt="hamburger-image">
</div>
```

3. In some sections of the HTML there are <p> elements inside <a> tags:
For example:
```
<a href="#header-text">
  <p id="home" class="hamburger-section-selection">Home</p>
</a>
```

I think that for semantics `<p>` elements should not be inside `<a>` tags. If so, perhaps a better approach could be to use spans and hrefs:

```
<a href="#header-text" class="hamburger-section-selection">Home</a>
```

**Feedback on Specific Requests:**
"Navigation bar clickable?"
I’m not entirely sure what this question refers to, but if there are issues with navigation links, perhaps start with checking that `<a>` tags do not contain `<p>` elements.

"Should the navigation be hidden on mobile?"
The navigation does seem to be hidden via media queries. If there's an issue it might has to do with the hamburger menu’s behavior—fixing the HTML as suggested above could perhaps help?

"How do I center my button?"
There are multiple buttons in the project, and their positioning sometimes depend on their parent elements. I think I need more details to provide specific feedback.

"How do I adjust an image?"
There are several images in the project, with different placements and sizes.  I think I need more details to provide specific feedback.

"Header image—how do I remove the green field underneath?"
I couldn't find a green field under the header image. More details would be helpful. Also, as mentioned earlier, making the header image part of a dedicated hero section could be a good approach for this assignment.

Conclusion:
The website is both responsive and playful—great work! 👌👌👌

tildetilde commented on index.html:
```diff
@@ -0,0 +1,263 @@
+      <h2>Meet us at the farm</h2>
+
+      <!-- hamburger navigation bar toggles on click with different icons (mobile/tablet) or show (desktop/widescreen) -->
+      <div
+        id="hamburger-menu-closed"
+        class="hamburger-menu-closed"
+      >
+        <img
+          id="hamburger-image"
+          class="hamburger-image"
+          src="./images/hamburger_filled.png"
+          alt="hamburger-image"
+        >
+      </div>
```
```suggestion
      <div id="hamburger-menu-closed" class="hamburger-menu-closed">
        <img id="hamburger-image" class="hamburger-image" src="./images/hamburger_filled.png" alt="hamburger-image">
      </div>
```

solen80a replied:
This is auto generated by VS code.

solen80a committed:
Added border styling to hamburger navigation. Cleaned the code and added some comment.

solen80a committed:
Cleaned out unused code.

solen80a committed:
Made formatting changes to the form to better fit mobile, tabler.

solen80a committed:
Fixed the formatting for desktop.

solen80a committed:
Fixed som comments

HIPPIEKICK reviewed (changes requested):
Nice job with this project Sofia! You’ve got almost all parts there, the only thing to have a look at again is that one of the requirements for this project was to use CSS Grid, and I can’t see that it’s being used, therefor I’m asking for changes.

Another thing I thought about that can be nice to think about going forward is the UX. Users are used that websites look and behave in a specific way, and I think that’s why I find your hamburger menu a bit confusing (unconventional position and how to close it is not super clear).

**Missing requirement**
- CSS Grid

solen80a commented:
Hi,

My articles are display: grid and my understanding is that they then are CSS Grid or have I missunderstod something?

Do you want me to change the hamburger for this site or do you refer to think about this in other projects?

Thanks Sofia

HIPPIEKICK commented:
> Hi,
> 
> My articles are display: grid and my understanding is that they then are CSS Grid or have I missunderstod something?
> 
> Do you want me to change the hamburger for this site or do you refer to think about this in other projects?
> 
> Thanks Sofia

As you can see in this image, your Grid styling is overridden by flexbox. Apart from this, I'd like to ask what you would like to accomplish by using CSS Grid here. Having a one-column grid doesn't really put CSS Grid to use. There are so many cool things you could accomplish with CSS Grid, but in this case it would work with `display: block;` so it doesn't really showcase CSS Grid if that makes sense? Let me know if you have further questions. Regarding the hamburger, it's up to you. Not a requirement 😊 
![Screenshot 2025-03-05 at 14 44 45](https://github.com/user-attachments/assets/a472e3f6-ca39-4d4e-9ac1-a77b5123572a)

solen80a committed:
Changed footer from flexbox to CSS Grid. Also removed style for articles due to dublicat styling class berry, they were the same and override.

solen80a committed:
Adjuseted the footer-navigation grid for mobile, tablet, desktop and widescreen.

solen80a commented:
Hi,
Ok, missed that I somehow managed to add styling to both navigation and the associated class. Removed the part that were duplicates.

Changed my class footer-navigation from flexbox to css grid and adjusted it for different media types.

solen80a committed:
Removed styling part for articles due to duplication.

HIPPIEKICK reviewed (approved):
Great job with the changes ⭐
